src/components/search/Search.js

```diff
@@ -0,0 +1,60 @@
+import { useEffect, useState } from "react";
+import { useNavigate,useLocation } from "react-router-dom";
+import styles from "./Search.module.css";
+export const Search = () => {
+  const [search, setSearch] = useState("");
+  const query = new URLSearchParams(useLocation().search);
+  const searchParams = query.get("search");
+  useEffect(() => {
+    if (
+      searchParams === "" ||
+      searchParams === null
+    ) {
+      setSearch("");
+    }
+  }, [searchParams]);
+
+  useEffect(() => {
```

Comment:
Could have written this in the above use state itself as the dependency array is the same and minor difference in functionality.

Reply:
corrected it, thanks